File: banking-service/src/main/java/com/finpilot/banking/service/AccountService.java
import com.finpilot.banking.dto.DashboardResponse;
import com.finpilot.banking.entity.User;

public interface AccountService {

    Account createAccount(Account account);

    Account getAccount(Long id);

    BigDecimal getBalance(Long id);

    DashboardResponse getDashboard(User user);
}   

File: frontend/src/services/accountService.js
import api from "./api";
import { API } from "../api/endpoints";

const accountService = {

  createAccount(data) {
    return api.post(API.CREATE_ACCOUNT, data);
  },

  getAccount(id) {
    return api.get(API.ACCOUNT(id));
  },

  getBalance(id) {
    return api.get(API.BALANCE(id));
  }

};

export default accountService;